### PR TITLE
ENCD-4108 audit matched

### PR DESCRIPTION
**Pull request by gabdank**
adjusting the audit for chip control to treat properly matched sets

### CODE REVIEW

File: src/encoded/audit/experiment.py [hunk 1]
```diff
@@ -133,12 +141,37 @@ def audit_experiment_chipseq_control_read_depth(value, system, files_structure):
                 target_investigated_as)
 
 
-def get_control_target_name(control_id, control_objects):
+def check_control_target(control_id, control_objects, bam_id, bam_type):
```
Comment (caseylitton) [inline src/encoded/audit/experiment.py]: This isn't a big deal, but I wouldn't change functions name with out good reason.  In this case get_control_target_name only exists in this file but the code reviewer needs to check that.  Also, based on the naming conventions a prefix of 'check' leads me to think the function is boolean.  

Just something to consider when updating code.  A better name might be get_control_target_failures?

File: src/encoded/audit/experiment.py [hunk 2]
```diff
@@ -96,34 +96,42 @@ def audit_experiment_chipseq_control_read_depth(value, system, files_structure):
                     and
                     derived_from.get('dataset') in controls_files_structures
                     and
-                    check_pipeline('ChIP-seq read mapping',
-                                    derived_from.get('@id'),
-                                    controls_files_structures[derived_from.get('dataset')])))
+                    (check_pipeline(
```
[the remark this answers is not in the document]
Comment (gabdank) [reply src/encoded/audit/experiment.py]: fixed

File: src/encoded/audit/experiment.py [hunk 3]
```diff
@@ -96,34 +96,42 @@ def audit_experiment_chipseq_control_read_depth(value, system, files_structure):
...
+                        yield f
+                else:
+                    control_depth = get_chip_seq_bam_read_depth(bam_file)
+                    if not control_depth:
+                        detail = (
+                            'Control {} file {} ' +
+                            'has no associated quality metric, preventing calculation of the read depth.').format(
+                                bam_file['output_type'],
+                                bam_file['@id'])
+                        yield AuditFailure('missing control quality metric', detail, level='WARNING')
```
[the remark this answers is not in the document]
Comment (gabdank) [reply src/encoded/audit/experiment.py]: probably you are right. I will make the change and launch a demo to see if it works as expected.

File: src/encoded/audit/experiment.py [hunk 4]
```diff
@@ -96,34 +96,42 @@ def audit_experiment_chipseq_control_read_depth(value, system, files_structure):
...
+                        return
+                    else:
+                        cumulative_read_depth += control_depth
+                        triple = (
```
[the remark this answers is not in the document]
Comment (gabdank) [reply src/encoded/audit/experiment.py]: fixed

File: src/encoded/audit/experiment.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (gabdank) [reply src/encoded/audit/experiment.py]: it broke the test, reverting to yield followed by return

[the remark this answers is not in the document]
Comment (caseylitton) [reply src/encoded/audit/experiment.py]: interesting, maybe I'll look into that a little.  everything else looks good.

**Pull request merged**